.gitignore

```diff
@@ -14,3 +14,4 @@ bin
 
 .idea/
 .vscode/
+vendor
```

Comment:
nit: this should be a local ignore?

Reply:
[github/gitignore/Go.gitignore](https://github.com/github/gitignore/blob/master/Go.gitignore) doesn't add vendor/ to its .gitignore file.
```bash
# Dependency directories (remove the comment below to include it)
# vendor/
```
[gitignore.io/api/go](https://www.gitignore.io/api/go) does add vendor/ to its .gitignore.
```bash
# Dependency directories (remove the comment below to include it)
# vendor/

### Go Patch ###
/vendor/
/Godeps/
```

Reply:
@wenqiq Normally it would be better to add this via a separate PR if it is controversial so this PR can proceed faster.

Anyway Antrea doesn't include vendor directory in the repo so at least it's harmless to add it here for the convenience of vendor mode developers. @Dyanngg Do you have strong opinion to remove it from this PR? If not, I'm going to merge it as is.

Reply:
I don't have a strong opinion for this. Merged.